src/controllers/sheltersController.js

```diff
@@ -87,18 +93,33 @@ const updateShelter = async (req, res, next) => {
 
 
 const deleteShelter = async (req, res, next) => {
+  log.debug("Calling deleteShelter...");
   //ShelterID is ON DELETE CASCADE, If a shelter is removed, those PET records associated with the ShelterID will be removed automatically
   await deleteShelterByID(req.params.shelterID)
     .then((dbResponse) => {
       if (dbResponse.affectedRows == 0) res.sendStatus(404);
       else if (dbResponse.affectedRows == 1) res.sendStatus(204);
       else {
-        console.log(res);
+        log.error(res);
         res.sendStatus(500);
       }
     })
     .catch((e) => {
-      console.log(e);
+      log.error(e);
+      res.sendStatus(500);
+      next(e);
+    });
+};
+
+const getPets = async (req, res, next) => {
+  log.debug("Calling getPets...");
+  await getAllPets(req.params.shelterID)
+    .then((dbResponse) => {
+      if (dbResponse.length == 0) res.sendStatus(404);
```

Comment:
I wonder if we would want to distinguish 'shelter not found' or 'shelter exists but has no pets yet' here. It depends on how we want to implement the frontend, but I imagine we may need a way to tell the difference? Like if the shelter id doesn't exist, we want to return an error or redirect the user to the not found page, but if the shelter simply doesn't have any pets yet we may want to just display an empty list and a create button. If we want to tell the difference we'll probably need to run a select query to the shelters table first in the model. Let me know what you think!

Reply:
I agree, I changed the code to return below.
 'shelter not found': --> 404, "Shelter not found"
 'shelter exists but has no pets yet' --> 200. [] (empty array)